rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -227,6 +245,25 @@ bool Player::is_paused() const
   return true;
 }
 
+double Player::get_rate() const
+{
+  if (clock_) {
+    return clock_->get_rate();
+  }
+  return 1.0;
+}
+
+bool Player::set_rate(double rate)
+{
+  if (rate <= 0.0) {
+    return false;
```

Comment:
@emersonknapp I thought about that it would be better to put this check
```cpp
if (rate <= 0) {
```
inside `clock_->set_rate(rate);` and return boolean from `set_rate(rate)`. i.e. API changes require.

Reply:
I would love to be able to run time in revers - do you think that makes sense as a possible future feature? I was thinking that the "rate > 0" requirement is a current limitation of the `Player` - rather than being a limitation of the `PlayerClock`

Reply:
So well, we are doing abstraction in player around clock. And if clock would be able to run in revers why not? \
It least I don't see any problem in Player class if it would be negative. Please correct me if I am wrong.